dlt-database/migrations/0001-init_db.ts

```diff
@@ -0,0 +1,121 @@
+/* FIRST MIGRATION
+ *
+ * This migration is special since it takes into account that
+ * the database can be setup already but also may not be.
+ * Therefore you will find all `CREATE TABLE` statements with
+ * a `IF NOT EXISTS`, all `INSERT` with an `IGNORE` and in the
+ * downgrade function all `DROP TABLE` with a `IF EXISTS`.
+ * This ensures compatibility for existing or non-existing
+ * databases.
+ */
+
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+export async function upgrade(queryFn: (query: string, values?: any[]) => Promise<Array<any>>) {
+  // write upgrade logic as parameter of queryFn
+  await queryFn(`
+    CREATE TABLE IF NOT EXISTS \`users\` (
+      \`id\` int(10) unsigned NOT NULL AUTO_INCREMENT,
+      \`gradido_id\` char(36) DEFAULT NULL,
+      \`derive1_pubkey\` binary(32) NOT NULL,
```

Comment:
very nice, now the derivation comes much more expressive simply from db-model...